docs/src/idiomatic/007if-then-else.md

```diff
@@ -0,0 +1,94 @@
+
+The `IF-THEN-ELSE` operator can be used either to define a value or to branch some action as a sort of flow control. Defining values with `IF-THEN-ELSE` is common practice and is similar to the use of `IF` expressions on declarative programming languages. However, flow control in TLA+ can be done naturally by behavior definition through actions, making the use of `IF-THEN-ELSE` for flow control unnecessary. This idiom aims to clarify different usages of `IF-THEN-ELSE` expressions keeping in mind the TLA+ essence of declaring actions to define sets of transitions. 
+
+## When to use `IF-THEN-ELSE`
+
+### When the result is not Boolean
+
+When the `IF-THEN-ELSE` expression doesn't evaluate to a boolean value, it cannot be rewritten using boolean operators, so this idiom doesn't apply.
+
+```tla
+SafeDiv(x, y) == IF y /= 0 THEN x/y ELSE 0
+```
+
+### When the result is a state formula
+
+State formulas are formulas that don't contain any action operator (e.g. primed variables, `UNCHANGED`). Using `IF-THEN-ELSE` on this type of formula can make it easier to read in some situations, and don't come with any disadvantage. This example state formula uses `IF-THEN-ELSE` to return a boolean value: 
+
+```tla
+ValidIdentity(person) == IF Nationalized(person) THEN ValidId(person) ELSE ValidPassport(person)
+```
+
+Although it could be rewritten with boolean operators, it doesn't read as nicely:
+
+```tla
+ValidIdentity(person) == \/ /\ Nationalized(person)
+                            /\ ValidId(person)
+                         \/ /\ !Nationalized(person)
+                            /\ ValidPassport(person)
+```
+
+## When (and how) *not* to use `IF-THEN-ELSE`
+
+Mixing `IF-THEN-ELSE` expressions with action operators introduces unnecessary branching to definitions that could be self-contained and look more like a transition definition. 
+
+```tla
+Withdraw(amount) == IF balance >= amount 
+                     THEN /\ balance' = balance - amount
+                          /\ response' = "SUCCESS"
+                     ELSE /\ UNCHANGED<balance>
+                          /\ response' = "FAILURE"
+```
+
+We could separate the two branches into their own actions with clarifying names and explicit conditions, and use a disjunction between the two actions instead of the `IF-THEN-ELSE` block:
+
+```tla
+WithdrawSuccess(amount) == /\ balance >= amount 
+                           /\ balance' = balance - amount
+                           /\ response' = "SUCCESS"
+                            
+WithdrawFailure(amount) == /\ balance < amount 
+                           /\ response' = "FAILURE"
+                           /\ UNCHANGED<balance> 
+                            
+Withdraw(amount) == WithdrawSuccess(amount) \/ WithdrawFailure(amount)
+```
+
+## Advantages
+
+- Each action declares fewer transitions, so it's easier to reason about it
+- A disjunction of actions is closer to a union of transition sets than an `IF-THEN-ELSE` expression
```

Comment:
I mean the set of transitions derived from an action

Reply:
I know what you meant, but as a general rule, I'd recommend caution with terminology, because, especially with TLA+, it's easy for someone to misunderstand this and assume there is a TLA+ expression/construct that is a literal set.
We have a lot of documentation (and papers) on the treatment of transition systems, full of terminology (e.g. https://apalache.informal.systems/docs/tutorials/symbmc.html), and I'd like to advocate against diverging on terminology in the docs. 
This might come across as nitpicking, but Igor and I have spent many hours discussing terminology in docs, and you become sensitive to this kind of stuff.

Reply:
Not at all, I'll totally agree with consistent terminology, I just didn't know/recall the standard term here, sorry! I've quickly scrolled through the doc but I'm still not sure about the best term here. Would you write "set of transitions"? Or should I change the whole phrase to address transitions as a relation?